src/js/components/AppListLabelsFilterComponent.jsx

```diff
@@ -144,13 +130,21 @@ var AppListLabelsFilterComponent = React.createClass({
   handleChange: function (label, event) {
     var state = this.state;
     var selectedLabels = [];
+    var update = React.addons.update;
+
+    var labelIndex = state.selectedLabels.findIndex(currentLabel => {
+      return currentLabel[0] === label[0] && currentLabel[1] === label[1];
+    });
 
     if (event.target.checked === true) {
-      selectedLabels = lazy(state.selectedLabels).union(label).value();
+      if (labelIndex === -1) {
+        selectedLabels = update(state.selectedLabels, {$push: [label]});
```

Comment:
Well not necessarily. Lazy's `union` also did not mutate the sate directly, so in that sense it's no different. It surely is more React-y but the final effect is unvaried.